waddrmgr/db.go

```diff
@@ -1865,3 +1824,166 @@ func upgradeToVersion3(namespace walletdb.Namespace, seed, privPassPhrase, pubPa
 	}
 	return nil
 }
+
+// upgradeToVersion4 upgrades the database from version 3 to version 4
+// The following fields were added to the serialized address row to support
+// imported public addresses:
+// * 'watchingOnly' - boolean indicating whether the address is watching-only
+// * 'encryptedPubKeyHash' - encrypted pubkey hash of imported p2pkh address
+func upgradeToVersion4(namespace walletdb.Namespace) error {
+	err := namespace.Update(func(tx walletdb.Tx) error {
+		bucket := tx.RootBucket().Bucket(addrBucketName)
+
+		err := bucket.ForEach(func(k, v []byte) error {
+			// Skip buckets.
+			if v == nil {
+				return nil
+			}
+
+			// Since importing watching-only addrs was not added
+			// until this version, mark all existing addresses as
+			// not being watching-only.
+			serializedAddress := make([]byte, len(v)+1)
+			copy(serializedAddress[:13], v[:13])
```

Comment:
This is an off-by-one.  It should be:
```Go
copy(serializedAddress[:14], v[:14])
```

I'm really concerned the tests are not catching these.